### DISCUSSION TITLE
Auto Close On Edit Form

### DISCUSSION THREAD

**Discussion by danicrosby** [nss-evening-cohort-14/evening-client-side#308] [Help Me!] [answered]
Dani Crosby

This seems like a very basic thing, but myself and others are having the same issue. 

I would like for the edit form to automatically close/collapse when the 'edit-submit-button' is clicked.
As it stands, I am only able to collapse the edit form by manually clicking the 'close-form-button' which is a different button from the 'edit-submit-button'.
I am able to click the 'add-form-submit' and have values saved, and then redirect to the homepage. All that works.
I am able to click on a single card and see all the form values.
I am able to click on an edit button and an edit form pops down where I can edit the add-form values in real-time. That all works.
I understand that the 'edit-submit-button' and the 'add-form-submit' are using the same logic and when those buttons 'submit' the form updates.
I assume there is an additional piece of logic needed on the 'edit-submit-button' to get that form to collapse, but I don't know how to implement it without also changing the functionality of the  'add-submit-button'.

No errors appear.
No API's used.

I have tried to put the edit form in a popover, modal and other various views.
I have tried to put additional logic on the edit submit button but this logic makes the add info portion wonky.
I have read and implemented several things from the REACTSTRAP documents.

https://reactstrap.github.io/components/form/
https://reactstrap.github.io/components/form/
https://stackoverflow.com/questions/56816001/how-to-redirect-collapse-an-expanded-edit-form-after-hitting-submit-in-react-adm
https://www.sitepoint.com/work-with-forms-in-react/

Github References:

Dream Form:
https://github.com/danicrosby/LUCID-Front-End-Capstone/blob/main/src/components/DreamForm.js

Single Dream Card with Edit Form:
https://github.com/danicrosby/LUCID-Front-End-Capstone/blob/main/src/components/SingleDreamCard.js

**Comment by drteresavasquez** [comment, marked as answer]
How do you get the form to open? I would assume that you could do the opposite of that to get it to close on the submit, but you will have to pass that function down to your form as a prop.

**Reply by drteresavasquez** [reply]
@danicrosby you have a state item called setEditing. you can pass that to the form and call it on the handleSubmit and set it to false. That should close your form on submit.
